src/components/encoding-pane/encoding-shelf.tsx

```diff
@@ -65,36 +59,17 @@ class EncodingShelfBase extends React.PureComponent<EncodingShelfProps, Encoding
     );
   }
 
-  // TODO: consider extracting this to another file
   private field() {
-    const {id, fieldDef} = this.props;
-
-    const caretHide = !(fieldDef.type === 'quantitative' || fieldDef.type === 'temporal');
-
-    // TODO: apply is over
-    // TODO(https://github.com/vega/voyager/issues/285): support clicking outside popup to disable
-
+    const {id, fieldDef, handleAction} = this.props;
     return (
-      <div styleName="field-wrapper">
-        <TetherComponent
-          attachment="top left"
-          targetAttachment="bottom left"
-        >
-          <Field
-            fieldDef={fieldDef}
-            caretOnClick={this.toggleFunctionPopup}
-            caretHide={caretHide}
-            isPill={true}
-            parentId={{type: FieldParentType.ENCODING_SHELF, id: id}}
-            draggable={true}
-            onRemove={this.onRemove}
-          />
-
-          { this.state.functionPopupOpen &&
-            <FunctionPicker fieldDef={fieldDef} onFunctionChange={this.onFunctionChange}/>
-          }
-        </TetherComponent>
-      </div>
+      <FieldWrapper
+        id={id}
+        fieldDef={fieldDef}
+        functionPopupOpen={this.state.functionPopupOpen}
```

Comment:
I think `functionPopupOpen`, `toggleFunctionPopup` and `closeFunctionPopup` should be maintained internally inside `FieldWrapper`?  

(Why should `EncodingShelf` maintain the state for its internal component?)